web/ui/react-app/src/pages/graph/Panel.tsx

```diff
@@ -138,55 +144,72 @@ class Panel extends Component<PanelProps, PanelState> {
         throw new Error('Invalid panel type "' + this.props.options.type + '"');
     }
 
-    fetch(`${this.props.pathPrefix}/${API_PATH}/${path}?${params}`, {
-      cache: 'no-store',
-      credentials: 'same-origin',
-      signal: abortController.signal,
-    })
-      .then(resp => resp.json())
-      .then(json => {
-        if (json.status !== 'success') {
-          throw new Error(json.error || 'invalid response JSON');
-        }
-
-        let resultSeries = 0;
-        if (json.data) {
-          const { resultType, result } = json.data;
-          if (resultType === 'scalar') {
-            resultSeries = 1;
-          } else if (result && result.length > 0) {
-            resultSeries = result.length;
-          }
+    let query;
+    let exemplars;
+    try {
+      query = await fetch(`${this.props.pathPrefix}/${API_PATH}/${path}?${params}`, {
+        cache: 'no-store',
+        credentials: 'same-origin',
+        signal: abortController.signal,
+      }).then(resp => resp.json());
+
+      if (query.status !== 'success') {
+        throw new Error(query.error || 'invalid response JSON');
+      }
+
+      if (showExemplars) {
+        exemplars = await fetch(`${this.props.pathPrefix}/${API_PATH}/query_exemplars?${params}`, {
```

Comment:
Hmm, we're sending an extra `step` parameter here, which currently doesn't cause a problem, but is a bit unclean, as it depends on the current accidental fact that the API doesn't return `400 Bad Request` errors when it encounters extra parameters. To still keep things simple and not require a new `URLSearchParams` object, how about just doing `params.delete('step');` before this line? Also kinda makes that explicit to the reader.